daffodil-io/src/main/scala/org/apache/daffodil/io/DirectOrBufferedDataOutputStream.scala

```diff
@@ -48,33 +48,31 @@ private[io] class ByteArrayOutputStreamWithGetBuf() extends java.io.ByteArrayOut
 
 private[io] class ByteArrayOrFileOutputStream(
   maxBufferSizeInBytes: Long,
-  tempFilePath: Path,
-  useFileOnly: Boolean)
-  extends java.io.OutputStream
-  with Logging {
-  val filePath = {
-    if (tempFilePath.toString == "")
-      Paths.get(System.getProperty("java.io.tmpdir"))
-    else
-      tempFilePath
-  }
+  tempDirPath: Path,
+  existingFile: Path)
+  extends java.io.OutputStream with Logging {
 
-  var stream: java.io.OutputStream = {
-    if (useFileOnly)
-      new java.io.FileOutputStream(File.createTempFile("daffodil-", ".tmp", filePath.toFile))
+  val dirPath = {
+    if (tempDirPath.toString == "")
+      Paths.get(System.getProperty("java.io.tmpdir"))
```

Comment:
Thoughts on moving this to the tunable logic? It feels to me like this class should always expect to receive a valid path, and the caller should be reponsible for figuring on that path. In our case, the caller value comes from tunables, and the tunable code generator should contain the logic for "tunable value of emptry string should become java.io.tmpdir".

Reply:
For reference, the change here would be done here in TunableGenerator:

https://github.com/apache/incubator-daffodil/blob/master/daffodil-propgen/src/main/scala/org/apache/daffodil/propGen/TunableGenerator.scala#L144-L153

You could create a custom TunableBase that sets the scalaConversion and scalaDefinition to values that have the empty string equals tmp dir logic, and then add a special case in the highlighted section in that link to create that.